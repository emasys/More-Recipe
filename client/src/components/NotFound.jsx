import React, { Component } from 'react';
import Navbar from './Navbar';
import { Link } from 'react-router-dom';

class NotFound extends Component {
  render() {
    return (
      <div className="container">
        <Navbar />
        <div className="text-center error-message mt-80">
          <div className="catalog-wrapper">
            <img src="../img/logo.png" alt="logo" />
            <h2 className="p-3 m-20">
              We Know this is embarrassing,<br /> but don't worry we won't judge you
            </h2>
            <h2>
              ...Go back{' '}
              <Link to="/" className="btn btn-dark hvr-icon-forward" role="button">
                Home&nbsp;
              </Link>{' '}
            </h2>
          </div>
        </div>
      </div>
    );
  }
}

export default NotFound;
