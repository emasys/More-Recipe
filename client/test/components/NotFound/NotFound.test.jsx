import React from 'react';
import { shallow } from 'enzyme';

// Component
import NotFound from '../../../src/components/NotFound';

let wrapper = null;

describe('Test suite for 404 page', () => {
  beforeEach(() => {
    wrapper = shallow(<NotFound />);
  });
  it('Should render without errors', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper.find('#error_msg').text()).toMatch('The content you are looking for does not exist');
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
