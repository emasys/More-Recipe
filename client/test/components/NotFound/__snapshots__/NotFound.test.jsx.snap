// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 404 page without errors 1`] = `
<div
  className="container"
>
  <Connect(Navbar)
    className="bg-dark fixed-top"
  />
  <div
    className=" row catalog-wrapper text-center justify-content-center  mt-80 mb-20"
  >
    <div
      className="col"
    >
      <img
        alt="logo"
        className="col-12 not-found-logo"
        src="https://res.cloudinary.com/emasys/image/upload/v1516439649/mR_2_jwnuce.png"
      />
      <h2
        className="p-3 m-20"
      >
        Oops...
      </h2>
      <h2
        className="p-3 m-20"
      >
        The content you are looking for does not exist
      </h2>
      <h2>
        <Link
          className="btn btn-dark btn-lg hvr-icon-back"
          replace={false}
          role="button"
          to="/"
        >
          ...Go back Home
        </Link>
      </h2>
    </div>
  </div>
</div>
`;
